### ISSUE TITLE
New error in "PEAR.ControlStructures.MultiLineCondition" sniff causes issues

### ISSUE THREAD

**Issue by aik099** [squizlabs/PHP_CodeSniffer#454]
The "SpacingAfterOpenBrace" error in "PEAR.ControlStructures.MultiLineCondition", that was introduced only in 2.2.0 release now checks that there must be no space between opening parenthesis of IF statement and 1st condition.

This conflicts with another sniff that allows for any X number of spaces (0 by default) that should be after opening parenthesis.

**Comment by gsherwood** [comment]
None of the included standards are conflicting That check was specifically added because it was missing from the PEAR standard and so wasn't fixing code correctly.

Are you using 2 sniffs in a custom standard that conflict? If so, which 2 sniffs and what settings? What code are you running the standard over?

**Comment by aik099** [comment]
Yes. It's `PSR2.ControlStructures.ControlStructureSpacing` sniff with `requiredSpacesAfterOpen` set to `1` plus this `PEAR.ControlStructures.MultiLineCondition`.

This way `ControlStructureSpacing` sniff wants 1 space after `if (`, but `MultiLineCondition` wants 0 space after `if (`. I guess this would end up in infinite fixing loop if I was fixing code.

**Comment by aik099** [comment]
I guess `Squiz` standard can be affected as well (if somebody wants to customize it).

**Comment by gsherwood** [comment]
You can't use the PEAR sniff if you want to enforce a single space after the braces. PEAR disallows this, so sniffs in there are going to enforce 0 spaces.

I think what you really need is an option in the PEAR sniff like the PSR sniff has. Does this sound right?

**Comment by aik099** [comment]
> I think what you really need is an option in the PEAR sniff like the PSR sniff has. Does this sound right?

Yes, option for specifying how much spaces needs to be after `(` and before `)` in control structure conditions is always a good idea. For now I've disabled that particular error in my `ruleset.xml`.

If PEAR has a rule, that no space should be after `if (` then more general control structure sniff should check that and not the `MultiLineCondition` sniff, which only cares about IFs with multi-line conditions.

**Comment by dereuromark** [comment]
The other issue is that valid PSR-2 code like this is now falsely reported:
```php
if (
    (count($this->_values) && $data instanceof Query) ||
    ($this->_query && is_array($data))
) {
    ...
}
```

Please revert this.
The PSR-2 states:

> Opening parentheses for control structures MUST NOT have a space after them, and closing parentheses for control structures MUST NOT have a space before.

That means a real "space", not a newline-whitespace.
Note that you do NOT check on closing parentheses and whitespace before that one either (which would currently be ok, and would be inconsistent to the breaking chance introduced in 2.2.0).

**Comment by gsherwood** [comment]
@dereuromark The change you are talking about is here: https://github.com/squizlabs/PHP_CodeSniffer/commit/ccf1f709f77fd8e0075d11c88b7f63816ccbb0ce

It is completely unrelated to the change @aik099 is talking about, which is only in the PEAR standard.

You may want to submit a different report for the issue you are having. Then we can have a discussion about if newlines are the same thing as spaces or not.

**Comment by dereuromark** [comment]
I am sorry if I confused the repos. If you want to can port the relevant information there.

**Comment by TomasVotruba** [comment]
Got same issue in 2.2.0 with negation in front:

`if ( ! $this->isDbSchemaPrepared) {`

 (PEAR.ControlStructures.MultiLineCondition.SpacingAfterOpenBrace)

**Comment by aik099** [comment]
In my `ruleset.xml` I've disabled that particular error altogether because check itself is misplaced (already one by other sniffs) as I've mentioned above.
